### PR TITLE
Add a standard dataset constructor.

### PR DESCRIPTION
**Pull request by jonthegeek**
And simplify the entailment vignette.

### CODE REVIEW

**Comment by jonthegeek** [comment]
I've been way too lax about test coverage. Gonna add tests before we merge this.

File: R/datasets.R [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    )
+
+    y <- .standardize_bert_dataset_outcome(y)
+
+    # We supply the data as tensors.
+    self$tokenized_text <- torch::torch_tensor(tokenized_text$token_ids)
+    self$token_types <- torch::torch_tensor(tokenized_text$token_type_ids)
+
+    # Also supply the labels as tensors.
+    self$y = torch::torch_tensor(as.integer(y))
```
Comment (jonathanbratt) [inline R/datasets.R:49]: Just a note while I'm thinking of it that the labels are often plagued by the 0/1-indexing issue, and eventually we should try to catch possible errors of this type.

**Review by jonathanbratt — COMMENTED** [review]
This is just a draft PR, so I won't submit an approving review yet. But it looks good so far!

File: R/datasets.R (hunk 1 above)
Comment (jonathanbratt) [reply R/datasets.R]: I hope so, but if I recall there is possibly some inconsistency in torch (at least in the documentation) about this.

From the documentation for `torch::nn_cross_entropy_loss()`:

> This criterion expects a class index in the range [0, C-1] as the target for each value of a 1D tensor of size minibatch; if ignore_index is specified, this criterion also accepts this class index (this index may not necessarily be in the class range).

I _think_ that the actual behavior is 1-indexed in this case, but I haven't checked all the loss functions.

Comment (jonthegeek) [reply R/datasets.R]: Ok, let's definitely watch for this as we use things! I think if something expects 0-indexed labels in {torch}, that's probably a {torch} bug, given the convention of the rest of the package.

**Review by jonathanbratt — APPROVED** [review]
looks good!

**Pull request merged**